gateway/api_definition.go

```diff
@@ -539,16 +539,25 @@ func (a APIDefinitionLoader) FromRPC(orgId string, gw *Gateway) ([]*APISpec, err
 
 func (a APIDefinitionLoader) processRPCDefinitions(apiCollection string, gw *Gateway) ([]*APISpec, error) {
 
-	var apiDefs []*nestedApiDefinition
-	if err := json.Unmarshal([]byte(apiCollection), &apiDefs); err != nil {
+	var payload []nestedApiDefinition
+	if err := json.Unmarshal([]byte(apiCollection), &payload); err != nil {
 		return nil, err
 	}
 
+	list := &nestedApiDefinitionList{
+		Message: payload,
+	}
+
+	gwConfig := a.Gw.GetConfig()
+
+	// Extract tagged entries only
+	apiDefs := list.filter(gwConfig.DBAppConfOptions.NodeIsSegmented, gwConfig.DBAppConfOptions.Tags...)
+
 	var specs []*APISpec
 	for _, def := range apiDefs {
 		def.DecodeFromDB()
```

Comment:
@furkansenharputlu this line smells a bit, the RPC is a request/response api - why do we need DecodeFromDB?

Reply:
@furkansenharputlu I just got your comment (was editing mine for more context). The RPC definitions are fed from a dashboard RPC response, so this function does not have a database use context.